File: min_max_number/min_max.py
#!/usr/bin/env python3

def min(nombres):
    if type(nombres) is list and len(nombres) > 0:
        if all(isinstance(i, (int, float)) for i in nombres):
            minimum = nombres[0]

            for nombre in nombres:
                if nombre < minimum:
                    minimum = nombre
            return minimum

    return -1


def max(nombres):
    if type(nombres) is list and len(nombres) > 0:
        if all(isinstance(i, (int, float)) for i in nombres):
            minimum = nombres[0]

            for nombre in nombres:
                if nombre > minimum:
                    minimum = nombre
            return minimum

    return -1


def min_max(nombres):
    return (min(nombres), max(nombres))


# 30 tests
tests = [
    ['a', 'b', 'c'],
    [1, 2, 3],
    [1.0, 2.0, 3.0],
    [1, 2.0, 3],
    [1, 3, "9"],
    [-5, 0, 2, 5, 3, -1]
]

for test in tests:
    print(min_max(test))

File: min_max_number/min_max.js
function min(tableau) {
  if (
    Array.isArray(tableau) &&
    tableau.every(function (element) {
      return typeof element === 'number';
    }) &&
    tableau.length > 0
  ) {
    let min = tableau[0];
    for (const nombre of tableau) {
      if (nombre < min) {
        min = nombre;
      }
    }
    return min;
  }

  return -1;
}

function max(tableau) {
  if (
    Array.isArray(tableau) &&
    tableau.every(function (element) {
      return typeof element === 'number';
    }) &&
    tableau.length > 0
  ) {
    let max = tableau[0];
    for (const nombre of tableau) {
      if (nombre > min) {
        max = nombre;
      }
    }
    return max;
  }

  return -1;
}

function min_max(tableau) {
  return [min(tableau), max(tableau)];
}

const arr1 = ['kjfjkdkjs', -1000, 2, -100];

console.log(min(arr1), max(arr1));
console.log(min_max(arr1));
